homeassistant/components/dio_chacon/entity.py

```diff
@@ -0,0 +1,53 @@
+"""Base entity for the Dio Chacon entity."""
+
+from dio_chacon_wifi_api import DIOChaconAPIClient
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER
+from .coordinator import DioChaconDataUpdateCoordinator
+
+
+class DioChaconEntity(CoordinatorEntity):
+    """Implements a common class elements representing the Dio Chacon entity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: DioChaconDataUpdateCoordinator,
+        target_id: str,
+        name: str,
+        model: str,
+        connected: bool,
+    ) -> None:
+        """Initialize Dio Chacon entity."""
+        super().__init__(coordinator)
+
+        self.dio_chacon_client: DIOChaconAPIClient = coordinator.dio_chacon_client
+
+        self._attr_available = connected
+        self._target_id = target_id
+        self._attr_unique_id = target_id
+        self._attr_name = name
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._target_id)},
+            manufacturer=MANUFACTURER,
+            name=name,
+            model=model,
+        )
+
+    @property
+    def coordinator_data(self):
+        """Return the coordinator data for received callback information by filtering on the correct device id."""
+        if self.coordinator.data and self.coordinator.data["id"] == self._target_id:
```

Comment:
Okay so wait, how much devices do you support? The coordinator would either need to have the full state of the integration or you could split coordinators per device, but since everything comes through one callback, having one coordinator makes sense. But currently it looks to not have the complete state, can you elaborate on that?

Reply:
I have 2 types of devices : covers and switches.
The callback only returns data relating to the device that generated the event (ex a pushed button).
In this implementation, the coordinator does not contain the states of all the devices.
Should I store in the coordinator.data, the state of all the devices instead of only the device state update that generated the event ?

Reply:
New implementation done : removed coordinator usage to simplify the code as stated that is not pertinent since the callback only returns per device state.